clippy_lints/src/types.rs

```diff
@@ -66,6 +66,34 @@ declare_clippy_lint! {
     "usage of `Box<Vec<T>>`, vector elements are already on the heap"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for use of `&[char]` anywhere in the code.
+    ///
+    /// **Why is this bad?** `char` represents a Unicode codepoint, not a
+    /// character. Usually users of `&[char]` want O(1) indexing on characters,
```

Comment:
The docs should also say something about *how* to index on characters (e.g. `unicode_segmentation`?)

Reply:
According to the issue, the correct way to index on characters is to use `&[&str]`. I can try to make that more clear in the documentation if that is correct.

Reply:
I was not talking about the type but the code.